test/type-error.js

```diff
@@ -0,0 +1,563 @@
+import test from 'ava';
+import avaRuleTester from 'eslint-ava-rule-tester';
+import rule from '../rules/type-error';
+
+const ruleTester = avaRuleTester(test, {
+	env: {
+		es6: true
+	}
+});
+
+const errors = [{
+	ruleId: 'type-error',
+	message: '`new Error()` is too unspecific for a type check, use `new TypeError()` instead.'
+}];
+
+ruleTester.run('type-error', rule, {
+	valid: [
+		`if (MrFuManchu.name !== 'Fu Manchu' || MrFuManchu.isMale === false) {
+			throw new Error('How cant Fu Manchu be Fu Manchu?');
+		}`,
+		`if (Array.isArray(foo) || ArrayBuffer.isView(foo)) {
+			throw new TypeError();
+		}`,
+		`if (wrapper.g.ary.isArray(foo) || wrapper.f.g.ary.isView(foo)) {
+			throw new TypeError();
+		}`,
+		`if (wrapper.g.ary(foo) || wrapper.f.g.ary.isPiew(foo)) {
+			throw new Error();
+		}`,
+		`if (Array.isArray()) {
+			throw new Error('Woohoo - isArray is broken!');
+		}`,
+		`if (Array.isArray(foo) || ArrayBuffer.isView(foo)) {
+			throw new CustomError();
+		}`,
+		`if (Array.isArray(foo)) {
+			throw new Error.foo();
+		}`,
+		`if (Array.isArray(foo)) {
+			throw new Error.foo;
+		}`,
+		`if (Array.isArray(foo)) {
+			throw new foo.Error;
+		}`,
+		`if (Array.isArray(foo)) {
+			throw new foo.Error('My name is Foo Manchu');
+		}`,
+		`if (Array.isArray(foo) || ArrayBuffer.isView(foo)) {
+			throw Error('This is fo FooBar', foo);
+		}`,
+		`if (Array.isArray(foo) || ArrayBuffer.isView(foo)) {
+			new Error('This is fo FooBar', foo);
+		}`,
+		`function test(foo) {
+			if (Array.isArray(foo) || ArrayBuffer.isView(foo)) {
+				return new Error('This is fo FooBar', foo);
+			}
+			return foo;
+		}`,
+		`if (Array.isArray(foo) || ArrayBuffer.isView(foo)) {
+			lastError = new Error('This is fo FooBar', foo);
+		}`,
+		`if (!isFinite(foo)) {
+			throw new TypeError();
```

Comment:
Can you add this test case too but throwing `new Error()`? That should be reported